shell/platform/windows/public/flutter_windows.h

```diff
@@ -129,17 +137,10 @@ FLUTTER_EXPORT FlutterDesktopEngineRef FlutterDesktopEngineCreate(
 // |engine| is no longer valid after this call.
 FLUTTER_EXPORT bool FlutterDesktopEngineDestroy(FlutterDesktopEngineRef engine);
 
-// Starts running the given engine instance and optional entry point in the Dart
-// project. If the entry point is null, defaults to main().
-//
-// If provided, entry_point must be the name of a top-level function from the
-// same Dart library that contains the app's main() function, and must be
-// decorated with `@pragma(vm:entry-point)` to ensure the method is not
-// tree-shaken by the Dart compiler.
+// Starts running the given engine instance.
 //
 // Returns false if running the engine failed.
-FLUTTER_EXPORT bool FlutterDesktopEngineRun(FlutterDesktopEngineRef engine,
-                                            const char* entry_point);
+FLUTTER_EXPORT bool FlutterDesktopEngineRun(FlutterDesktopEngineRef engine);
```

Comment:
Do we have any "major" users of the desktop C API that would be affected by this breaking change? I feel uneasy with this, personally I'd prefer keeping the old API but marked as something.

FYI this will break @timsneath's Dart runner: https://github.com/timsneath/win32_runner/blob/3deb30a5de7594b66fffac08cdb292da8f7ab58e/lib/src/ffi.dart#L645

Reply:
One alternative would be to keep `RunWithEntrypoint` and have that either override `DartProject`'s entrypoint, or, fail if `DartProject` has an entrypoint. That would let us add this better API without breaking existing apps.

Reply:
I'm aware of @timsneath's and @knopp's embedders.

I really wish we'd named this `FlutterDesktopEngineRunWithEntrypoint` and then we could have used `FlutterDesktopEngineRun` for the new replacement and marked that `FLUTTER_DEPRECATE("insert nice warning here")`.

I actually do have a patch that adds the new alternative, and deprecates the old one (and adds a macro definition for `FLUTTER_DEPRECATED`), then yells at runtime and fails to launch if you pass both a custom entrypoint in _both_ the `FlutterDesktopEngineProperties` and via the parameter, otherwise uses whichever one you set.

That approach is nicer from a compatibility/migration point of view (though a little ugly maintenance wise); if we can think of a nice name for the new `FlutterDesktopEngineRun`, I'm good with landing that and specifying a removal of the old API in some Future Flutter 4.0 or so.

Reply:
Alright -- changed my mind. I've cleaned up the other patch to a state that's fully backward-compatible while not being miserable to maintain. Thanks for pushing back on this; the resulting patch I think gets us to a place where we have the fixture testing I wanted, fixes https://github.com/flutter/flutter/issues/93537, and preserves backward compatibility.

We can deal with the more contentious issue of how we deprecate the entrypoint parameter on `FlutterDesktopEngineRun` in the C API and FlutterEngine::Run` in the C++ client wrapper separately. I've filed https://github.com/flutter/flutter/issues/109285 to track that.